Scripts/prepare_release.py

```diff
@@ -213,11 +213,12 @@ def comment(self):
 
 class Release:
     REMOTE = "origin"
-    MASTER_BRANCH = "refs/heads/master"
     NEWSFRAGMENTS_PATH = "docs/dev/news"
 
-    def __init__(self, version: str) -> None:
+    def __init__(self, version: str, source_branch: str) -> None:
         self.version = version
+        self.source_branch_name = source_branch
+        self.source_branch_path = f"refs/heads/{self.source_branch_name}"
         self.branch_name = f"release-{version}"
         self.branch_path = f"refs/heads/{self.branch_name}"
```

Comment:
Since we're introducing a new set of branch name/paths, we should rename the existing ones to `release_branch_{name,path}` or something along those lines in order to avoid confusion.